sound/soc/sof/sof-audio.c

```diff
@@ -301,6 +302,24 @@ static int sof_setup_pipeline_connections(struct snd_sof_dev *sdev,
 		}
 	}
 
+	/*
+	 * During route setup above, playback stream routes are set up by going
+	 * through sink path, capture stream routes are set up by going through
+	 * source path. Obviously, feedback route from a widget in the capture
+	 * stream to another widget in the playback stream is not able to be setup.
+	 *
+	 * The feedback route is setup here if widgets on both sides of the route
+	 * are active.
+	 */
+	list_for_each_entry(sroute, &sdev->route_list, list)
+		if (sroute->src_widget->use_count && sroute->sink_widget->use_count &&
+		    !sroute->setup) {
+			ret = sof_route_setup(sdev, sroute->src_widget->widget,
+					      sroute->sink_widget->widget);
+			if (ret < 0)
+				return ret;
+		}
```

Comment:
This makes sense. Does the route free works when one of the stream is stopped?

Reply:
yes, it should When a widget is deleted, all its routes are disconnected first